build: add version stamp to release output

DevVersion opened the pull request:
* Adds a version stamp to the `package.json` in the release output. The version stamp consists of the current `HEAD` commit sha, and the branch name.

Closes #12407

jelbourn commented on tools/package-tools/package-version-stamp.ts:
```diff
@@ -0,0 +1,30 @@
+export function insertPackageJsonVersionStamp(packageJsonPath: string) {
+  const packageJson = require(packageJsonPath);
+
+  packageJson['_gitCommitStamp'] = getCurrentCommitSha();
```
Any reason to prefix with an underscore?

DevVersion replied:
No real strong reason. I just wanted to indicate that these properties are not relevant for most developers. I took `underscore` because NPM also does that for their metadata in `package.json`.

![image](https://user-images.githubusercontent.com/4987015/43413927-adda6134-9431-11e8-86db-c551eae771cd.png)

Let me know if you want me to remove it.

jelbourn replied:
I would prefer to remove it if `npm` uses it for their own internal properties

DevVersion replied:
Sure. I assume the naming itself is fine then? e.g. `gitCommitStamp`.

jelbourn commented on tools/package-tools/package-version-stamp.ts:
```diff
@@ -0,0 +1,30 @@
+  const packageJson = require(packageJsonPath);
+
+  packageJson['_gitCommitStamp'] = getCurrentCommitSha();
+  packageJson['_gitBranchStamp'] = getCurrentBranchName();
```
I thought it might also be useful to stamp the current git username, but I want to check with Igor/Brad about doing that. I'll let you know.

DevVersion commented:
@jelbourn Addressed your feedback. We could probably need a `spawnSync` utility function with `cwd` set to the project dir.. but not sure how much benefit we'll have from it. I like being explicit here.

<img width="435" alt="ca431cff3a7c2c29b8e5c294c3cbd3cf" src="https://user-images.githubusercontent.com/4987015/43419344-ef76fc7e-9440-11e8-88be-4b4480da5023.png">
